izpack-installer/src/main/java/com/izforge/izpack/installer/bootstrap/InstallerGui.java

```diff
@@ -124,19 +124,32 @@ public void run()
 	            });
 	        }
 	        
-	        if (langCode == null)
-	        {
-	          installerContainer.getComponent(LanguageDialog.class).initLangPack();
-	        }
-	        else
-	        {
-	          installerContainer.getComponent(LanguageDialog.class).propagateLocale(langCode);
-	        }
-	        if (!installerContainer.getComponent(RequirementsChecker.class).check())
-	        {
-	            logger.info("Not all installer requirements are fulfilled.");
-	            installerContainer.getComponent(Housekeeper.class).shutDown(-1);
-	        }
+                final Exception[] guiThreadEx = new Exception[1];
+                SwingUtilities.invokeAndWait(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (langCode == null)
+                        {
+                            try {
+                                installerContainer.getComponent(LanguageDialog.class).initLangPack();
+                            } catch (Exception ex) {
+                                guiThreadEx[0] = ex;
```

Comment:
It is not that hardly confused as you might think :-)

A very similar issue has been already resolved earlier:
- https://izpack.atlassian.net/browse/IZPACK-1185
- https://github.com/izpack/izpack/pull/411
in [this commit](https://github.com/izpack/izpack/pull/411/commits/904541effcc9403708ef3e6a0e971d1d3e489685#diff-0bb7d7048a9a69ae9aa512730aefacf0R114).

The language dialog issue has been just additionally introduced.

I understand your intention, the question was mainly about the array. I still can't see the need of the workaround using the array, because the case you mentioned as reason won't occur, right?
The "TODO" comment we can simply remove if that has been reviewed and we are sure now.

OK?